x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/logic/bulk_actions/rule_params_modifier.ts

```diff
@@ -22,22 +23,68 @@ export const deleteItemsFromArray = <T>(arr: T[], items: T[]): T[] => {
   return arr.filter((item) => !itemsSet.has(item));
 };
 
+// Check if current params have a configured data view id
+// and the action is not set to overwrite data views
+const isDataViewExistsAndNotOverriden = (
+  ruleParams: IndexPatternRuleParams,
+  action: BulkActionEditPayloadIndexPatterns
+) => ruleParams.dataViewId != null && !action.overwrite_data_views;
+
+// Check if the index patterns added to the rule already exist in it
+const isAddedIndexPatternsAlreadyExist = (
```

Comment:
I recommend considering the following renaming as it should simplify code reading

* `isAddedIndexPatternsAlreadyExist` -> `hasIndexPatterns`
* `isDeletedIndexPatternsNonExistent` -> `hasNotIndexPatterns`

Reply:
Renamed